Index: postgresql-9.6.2/src/backend/storage/buffer/bufmgr.c
===================================================================
--- postgresql-9.6.2.orig/src/backend/storage/buffer/bufmgr.c	2017-02-21 12:13:30.010029132 +0000
+++ postgresql-9.6.2/src/backend/storage/buffer/bufmgr.c	2017-03-11 10:26:32.763244489 +0000
@@ -128,6 +128,7 @@
  */
 int			target_prefetch_pages = 0;
 
+int         pgstudy_usleep = 0;
 /* local state for StartBufferIO and related functions */
 static BufferDesc *InProgressBuf = NULL;
 static bool IsForInput;
@@ -787,7 +788,7 @@
 
 			if (forkNum == MAIN_FORKNUM)
 			{
-				printf("[%d, %d, %d]\n", smgr->smgr_rnode.node.relNode, blockNum, *hit); fflush(stdout); usleep(1000);
+				printf("[%d, %d, %d]\n", smgr->smgr_rnode.node.relNode, blockNum, *hit); fflush(stdout); usleep(pgstudy_usleep * 1000);
 			}
 			return BufferDescriptorGetBuffer(bufHdr);
 		}
@@ -967,7 +968,7 @@
 
 	if (forkNum == MAIN_FORKNUM)
 	{
-		printf("[%d, %d, %d]\n", smgr->smgr_rnode.node.relNode, blockNum, *hit); fflush(stdout); usleep(1000);
+		printf("[%d, %d, %d]\n", smgr->smgr_rnode.node.relNode, blockNum, *hit); fflush(stdout); usleep(pgstudy_usleep * 1000);
 	}
 	return BufferDescriptorGetBuffer(bufHdr);
 }
Index: postgresql-9.6.2/src/backend/utils/misc/guc.c
===================================================================
--- postgresql-9.6.2.orig/src/backend/utils/misc/guc.c	2017-02-06 21:45:25.000000000 +0000
+++ postgresql-9.6.2/src/backend/utils/misc/guc.c	2017-03-11 10:24:08.130960145 +0000
@@ -401,6 +401,8 @@
 extern const struct config_enum_entry sync_method_options[];
 extern const struct config_enum_entry dynamic_shared_memory_options[];
 
+extern int         pgstudy_usleep;
+
 /*
  * GUC option variables that are exported from this module
  */
@@ -2798,8 +2800,7 @@
 		NULL, NULL, NULL
 	},
 
-	{
-		{"gin_pending_list_limit", PGC_USERSET, CLIENT_CONN_STATEMENT,
+	{ {"gin_pending_list_limit", PGC_USERSET, CLIENT_CONN_STATEMENT,
 			gettext_noop("Sets the maximum size of the pending list for GIN index."),
 			NULL,
 			GUC_UNIT_KB
@@ -2809,6 +2810,17 @@
 		NULL, NULL, NULL
 	},
 
+	{
+		{"pgstudy_usleep", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop(""),
+			NULL,
+			GUC_UNIT_MS
+		},
+		&pgstudy_usleep,
+		0, 0, 1000,
+		NULL, NULL, NULL
+	},
+
 	/* End-of-list marker */
 	{
 		{NULL, 0, 0, NULL, NULL}, NULL, 0, 0, 0, NULL, NULL, NULL
